### PR TITLE
Bind only non-NULL BLOB values as BLOB on oci8 and ibm_db2

### LINKED ISSUE #5359: Flush an object with NULL values in LOB fields leads to an error, OCI8
### Bug Report

|    Q        |   A
|------------ | ------
| BC Break    | yes/no
| ORM Version     | 2.11.2 
| DBAL Version     | 3.3.3  
| Database Driver| Oracle, OCI8
| PHP | 8.1.3 

Link to issue in "ORM" repository:  [#9601](https://github.com/doctrine/orm/issues/9601)

#### Summary

When working with optional fields in entity that have a BLOB type (blob, clob, text, ..., etc.), we get a warning:

`Deprecated: OCILob::writeTemporary(): Passing null to parameter #1 ($data) of type string is deprecated in vendor/doctrine/dbal/src/Driver/OCI8/Statement.php on line 76`
 
The error occurs when working with Oracle (OCI8)

#### How to reproduce

```xml
<?xml version="1.0" encoding="utf-8"?>
<doctrine-mapping xmlns="http://doctrine-project.org/schemas/orm/doctrine-mapping" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://doctrine-project.org/schemas/orm/doctrine-mapping https://www.doctrine-project.org/schemas/orm/doctrine-mapping.xsd">
  <entity name="ExampleProcess" table="JOB_PROCESS">
    <id name="id" type="integer" column="id">
      <generator strategy="SEQUENCE"/>
      <sequence-generator sequence-name="PROCESS_ID_SEQ" allocation-size="1" initial-value="1"/>
    </id>
    <field name="response" type="blob" column="response" nullable="true"></field>
  </entity>
</doctrine-mapping>
```

```php
<?php

class ExampleProcess
{
    private int $id;
    /**
     * @var null|string|resource
     */
    private $response;  // << optional field

    public function getId(): int
    {
        return $this->id;
    }
 

    public function getResponse(): string|null
    {
        if (\is_resource($this->response)) {
            return (string) stream_get_contents($this->response);
        }

        return $this->response ?? null;
    }

    public function setResponse(string|null $response): self
    {
        $this->response = $response;

        return $this;
    }
 
}

```

Example:

```php
<?php

$test = new ExampleProcess();
$em->persist($test);
$em->flush(); // this will trigger a warning

$test2 = new ExampleProcess();
$test2->setResponse(null);
$em->persist($test2);
$em->flush(); // this will trigger a warning

// unwanted code that allows you to temporarily bypass the problem:
$test3 = new ExampleProcess();
$test3->setResponse("");
$em->persist($test3);
$em->flush(); // this will work fine

```
 
#### Expected behavior

Optional fields are expected to allow NULL values

### PR DESCRIPTION
**Pull request by morozov**
Fixes #5359.

### CODE REVIEW

File: tests/Functional/BlobTest.php [hunk 1]
```diff
@@ -46,6 +46,29 @@ public function testInsert(): void
         self::assertEquals(1, $ret);
     }
 
+    public function testInsertNull(): void
+    {
+        if (TestUtil::isDriverOneOf('ibm_db2')) {
+            self::markTestIncomplete('The ibm_db2 driver requires additional rework for this test to pass');
```
Comment (morozov) [inline tests/Functional/BlobTest.php]: https://github.com/doctrine/dbal/pull/5467 should be fixed in order to unblock fixing this issue.

**Review by greg0ire — APPROVED** [review]

File: src/Driver/OCI8/Statement.php [hunk 2]
```diff
@@ -72,10 +72,14 @@ public function bindParam($param, &$variable, $type = ParameterType::STRING, $le
...
-            $lob = oci_new_descriptor($this->connection, OCI_D_LOB);
-            $lob->writeTemporary($variable, OCI_TEMP_BLOB);
+            if ($variable !== null) {
+                $lob = oci_new_descriptor($this->connection, OCI_D_LOB);
+                $lob->writeTemporary($variable, OCI_TEMP_BLOB);
 
-            $variable =& $lob;
+                $variable =& $lob;
+            } else {
+                $type = ParameterType::STRING;
```
Comment (fabacino) [inline src/Driver/OCI8/Statement.php:81]: Just out of curiosity, why did you choose `ParameterType::STRING` here over `ParameterType::NULL`?

Comment (morozov) [reply src/Driver/OCI8/Statement.php:81]: There’s no special handling of the null parameters in oci8.

Comment (fabacino) [reply src/Driver/OCI8/Statement.php:81]: I see, thanks for explaining.

**Pull request merged**